packages/vitest/src/utils/coverage.ts

```diff
@@ -1,120 +1,195 @@
 import { readFileSync, writeFileSync } from 'node:fs'
 import { relative } from 'pathe'
+import mm from 'micromatch'
+import type { parseModule as MagicastParseModule } from 'magicast'
```

Comment:
Is this correctly bundled? I don't want to have a big jump in size here since there is no tree shaking for dts files

Reply:
Good catch. The `dist/coverage.d.ts` seems to increase from 4000 characters to 67000 characters. 😄

Reply:
Fixed by moving all `magicast` logic to providers and removed `magicast` from `vitest` package's `devDependencies`. There's now a generic `{ read, write }` interface for this.